judgels-frontends/raphael/src/routes/uriel/contests/ContestsPage/ContestsPage.tsx

```diff
@@ -5,46 +5,89 @@ import { parse } from 'query-string';
 
 import Pagination from 'components/Pagination/Pagination';
 import { Card } from 'components/Card/Card';
+import SearchBox from 'components/SearchBox/SearchBox';
 import { Contest, ContestCreateData, ContestsResponse } from 'modules/api/uriel/contest';
 
 import { LoadingContestCard } from '../ContestCard/LoadingContestCard';
 import { ContestCard } from '../ContestCard/ContestCard';
 import { ContestCreateDialog } from '../ContestCreateDialog/ContestCreateDialog';
 import { contestActions as injectedContestActions } from '../modules/contestActions';
 
+import './ContestPage.css';
+
 export interface ContestsPageProps extends RouteComponentProps<{ name: string }> {
   onGetContests: (name?: string, page?: number) => Promise<ContestsResponse>;
   onCreateContest: (data: ContestCreateData) => Promise<Contest>;
 }
 
+interface ContestsFilter {
+  name?: string;
+}
+
 export interface ContestsPageState {
   response?: ContestsResponse;
+  filter?: ContestsFilter;
+  isSearchBoxLoading?: boolean;
 }
 
 class ContestsPage extends React.Component<ContestsPageProps, ContestsPageState> {
   private static PAGE_SIZE = 20;
 
   state: ContestsPageState = {};
 
+  componentWillMount() {
+    const queries = parse(this.props.location.search);
+    const name = queries.name as string;
+
+    this.setState({ filter: { name }, isSearchBoxLoading: false });
+  }
+
   render() {
     return (
       <Card title="Contests">
-        {this.renderCreateDialog()}
+        {this.renderHeader()}
         {this.renderContests()}
         {this.renderPagination()}
       </Card>
     );
   }
 
-  private renderCreateDialog = () => {
-    const { response } = this.state;
-    if (!response) {
-      return null;
-    }
-    const { config } = response;
-    if (!config.canAdminister) {
```

Comment:
Because it looks weird if the button suddenly missing when we searching.

I think when we are searching for something, we expect only the data will be changed, not the button or search box.

If we want to preserve this behavior, the search box should be missing as well (otherwise it will looks even weirder if the whole thing missing once search button is pressed)

Reply:
I see. Then I think we can just clear the `data` (not the whole `response`) when we click the search box, so that we just use the old `config` while searching. What do you think

Reply:
sounds reasonable

Reply:
How about we just use `isSearchBoxLoading` state to check if we need to render the loading bar?